src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -296,27 +317,28 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       stateData.startMode.get.start(this)
   }
 
-  private def callCompletedWhileAborting(callKey: OutputKey, outputs: CallOutputs, returnCode: Int): State = {
-    awaitCallComplete(callKey, outputs, returnCode)
-    val nextState = if (isWorkflowAborted) scheduleTransition(WorkflowAborted) else stay()
-    val updatedData = stateData.addPersisting(callKey)
-    nextState using updatedData
-  }
-
   when(WorkflowAborting) {
     case Event(message @ CallCompleted(callKey, outputs, returnCode), data) if callKey.isPersistedRunning(data) =>
-      callCompletedWhileAborting(callKey, outputs, returnCode)
+      handleCallCompleted(callKey, outputs, returnCode, message)
     case Event(message @ CallCompleted(collectorKey: CollectorKey, outputs, returnCode), data) if !collectorKey.isPending(data) =>
-      // Collector keys don't have to be in Running state.
-      callCompletedWhileAborting(collectorKey, outputs, returnCode)
-    case Event(CallAborted(callKey), data) if callKey.isPersistedRunning(data) =>
-      persistStatus(callKey, ExecutionStatus.Aborted, None)
+      // Collector keys are weird internal things and never go to Running state.
+      handleCallCompleted(collectorKey, outputs, returnCode, message)
+    case Event(message @ CallAborted(callKey), data) if callKey.isPersistedRunning(data) =>
+      persistStatusThenAck(callKey, ExecutionStatus.Aborted, sender(), message, None)
       val updatedData = data.addPersisting(callKey)
-      (if (isWorkflowAborted) scheduleTransition(WorkflowAborted) else stay()) using updatedData
-    case Event(CallFailed(callKey, returnCode, failure), data) if callKey.isPersistedRunning(data) =>
-      persistStatus(callKey, ExecutionStatus.Failed, returnCode)
+      if (isWorkflowAborted) scheduleTransition(WorkflowAborted)
+      stay() using updatedData
+    case Event(message @ CallFailed(callKey, returnCode, failure), data) if callKey.isPersistedRunning(data) =>
+      persistStatusThenAck(callKey, ExecutionStatus.Failed, sender(), message, returnCode)
       val updatedData = data.addPersisting(callKey)
-      (if (isWorkflowAborted) scheduleTransition(WorkflowAborted) else stay()) using updatedData
+      if (isWorkflowAborted) scheduleTransition(WorkflowAborted)
+      stay() using updatedData
+    case Event(PersistenceCompleted(callKey, executionStatus), data) if executionStatus == ExecutionStatus.Done =>
```

Comment:
Can you do...?:
case Event(PersistenceComplete(callKey, ExecutionStatus.Done) =>